Sources/Spezi/Module/Capabilities/Communication/CollectPropertyWrapper.swift

```diff
@@ -0,0 +1,73 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+
+import XCTRuntimeAssertions
+
+
+/// Refer to the documentation of ``Component/Collect``.
+@propertyWrapper
+public class _CollectPropertyWrapper<Value> {
+    // swiftlint:disable:previous type_name
+    // We want the type to be hidden from autocompletion and documentation generation
+
+    private var injectedValues: [Value]? // swiftlint:disable:this discouraged_optional_collection
+
+    /// Access the collected values.
+    /// - Note: The property is only accessible within the ``Component/configure()`` method.
+    public var wrappedValue: [Value] {
+        guard let values = injectedValues else {
+            preconditionFailure("""
+                                Tried to access @Collect for value [\(Value.self)] which wasn't injected yet. \
+                                Are you sure that you are only accessing @Collect within the `Component/configure` method?
+                                """)
+        }
+
+        return values
+    }
+
+    /// Initialize a new `@Collect` property wrapper.
+    public init() {}
+}
+
+/// A adopter of this protocol is a property of a ``Component`` that provides mechanisms to retrieve
```

Comment:
Not sure if this is in the right place here?

Reply:
Upise 🙈